ComfortZoneApp/app/src/main/java/com/example/comfortzone/MainActivity.java

```diff
@@ -29,14 +30,15 @@ public class MainActivity extends AppCompatActivity {
     private FusedLocationProviderClient fusedLocationClient;
 
     public static final String TAG = "Main Activity";
+    public static final int PERMISSION_ID = 44;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
         fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
-        getLastLocation(this, this, fusedLocationClient);
+        getLastLocation(this, fusedLocationClient);
```

Comment:
why not only request permission first in `onCreate` if needed? since we are not getting the location here, it's weird to use this method. We should only call `getLastLocation` in places where we actually need the location. I would recommend extract out the get permission logic and put it in this activity.